msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -45,26 +46,65 @@ Users must not enter a parameter for a header or cookie. If you need to pass hea
+* You can use [Adaptive Card Designer](https://adaptivecards.io/designer/) to bind the API response to the response rendering template and preview the Adaptive Card. Insert the template in the **CARD PAYLOAD EDITOR** and insert the sample response entry in the **SAMPLE DATA EDITOR**.
+
+  :::image type="content" source="../assets/images/Copilot/api-me-sbs-adaptive-card-designer.png" alt-text="Screenshots shows the Adaptive Card designer with the Adaptive Card template and the sample data.":::
+
 * Define the schema reference URL in the `$schema` property.
-* Define `jsonPath` as the path to the relevant data/array in API response. if the path points to an array, then each entry in the array will be a separate result and if the path points to an object, there will only be a single result. *[Optional]*
+* A `jsonPath` is recommended for arrays or when the data for the Adaptive Card isn't the root object. For example, if your data is nested under `productDetails`, your JSON path would be `productDetails`.
+* Define `jsonPath` as the path to the relevant data/array in API response. If the path points to an array, then each entry in the array binds with the Adaptive Card template and return as separate results. *[Optional]*
 * The supported values for `responseLayout` are `list` and `grid`.
 
-The `JsonPath` property in response rendering template is $ to indicate the root object of the response data is used to render the Adaptive Card, and you can update the `jsonPath` property to point another property in response data.
-
-If the root object of the OpenAPI schema contains well-known array property name, then Teams Toolkit uses the array property as root element to generate an Adaptive Card, and the array property name is used as `JsonPath` property for response rendering template. For example, if the property name contains `result`, `data`, `items`, `root`, `matches`, `queries`, `list`, `output` and the type is `array`, then it's used as root element.
+If the root object of the OpenAPI schema contains well-known array property name, then Teams Toolkit uses the array property as root element to generate an Adaptive Card, and the array property name is used as `JsonPath` property for response rendering template. For example, if the property name contains `result`, `data`, `items`, `root`, `matches`, `queries`, `list`, or `output` and the type is `array`, then it's used as root element.
+
+The following is a JSON example for a list of products to create a card result for each entry:
+
+```json
+{
+   "version": "1.0",
+   "title": "All Products",
+   "warehouse": {
+      "products": [
+        {
+          "id": "1",
+          "name": "Product 1",
+          "price": "$10"
+        },
+        {
+          "id": "2",
+          "name": "Product 2",
+          "price": "$20"
+        }
+      ]
+   }
+}
+```
+
+The array of results is under `products`, nested under `warehouse`, so the JSON path is `warehouse.products`.
```

Comment:
The example is  available in the [overview](https://learn.microsoft.com/en-us/microsoftteams/platform/messaging-extensions/api-based-overview#response-rendering-template) article.

Reply:
Maybe a link to the overview article might be helpful here. It would help show the end-to-end of what going from a JSON to an response template looks like.